tests/unit/test_checksum.cpp

```diff
@@ -112,6 +113,49 @@ uint16_t CalculateChecksum(const Ip6::Address &aSource,
     return CalculateChecksum(&data, sizeof(PseudoHeader) + payloadLength);
 }
 
+uint16_t CalculateChecksum(const Ip4::Address &aSource,
+                           const Ip4::Address &aDestination,
+                           uint8_t             aIpProto,
+                           const Message &     aMessage)
+{
+    // This method calculates the checksum over an IPv6 message.
+
+    enum : uint16_t
+    {
+        kMaxPayload = 1024,
```

Comment:
Can use `static constexper uint16_t kMaxPayload = 1024;` instead?

I think we did use `enum` for compile-time constants before transition to C++11, so there may be instances of this still in the code but we can use `constexpr` for this (which has the advantage of having explicit type).